### PR TITLE
Commander: Add reporting of preflight and prearming state

### PR DESCRIPTION
**Pull request by LorenzMeier**
This is essential for the operator to know if the system is ready to fly.

### CODE REVIEW

File: src/modules/commander/Commander.cpp [hunk 1]
```diff
@@ -2238,6 +2238,16 @@ Commander::run()
 
 			/* publish vehicle_status_flags */
 			status_flags.timestamp = hrt_absolute_time();
+
+			// Evaluate current prearm status
+			if (!armed.armed) {
+				bool preflight_check_res = PreFlightCheck::preflightCheck(nullptr, status, status_flags, true, true, true, 30_s);
```
Comment (bkueng) [inline src/modules/commander/Commander.cpp]: This should be:
```suggestion
				bool preflight_check_res = PreFlightCheck::preflightCheck(&mavlink_log_pub, status, status_flags, _arm_requirements.global_position, false, true,
				       hrt_elapsed_time(&_boot_timestamp))
```

File: src/modules/commander/Commander.cpp [hunk 2]
```diff
@@ -2238,6 +2238,16 @@ Commander::run()
...
+				bool prearm_check_res = PreFlightCheck::preArmCheck(nullptr, status_flags, safety_s{},
```
Comment (bkueng) [inline src/modules/commander/Commander.cpp]:
```suggestion
				bool prearm_check_res = PreFlightCheck::preArmCheck(nullptr, status_flags, _safety,
```

File: src/modules/commander/Commander.cpp [hunk 3]
```diff
@@ -2238,6 +2238,16 @@ Commander::run()
...
+							PreFlightCheck::arm_requirements_t{}, status);
```
Comment (bkueng) [inline src/modules/commander/Commander.cpp]:
```suggestion
							_arm_requirements, status);
```

File: src/modules/commander/Commander.cpp (hunk 1 above)
Comment (MaEtUgR) [reply src/modules/commander/Commander.cpp]: I think the idea was that if you don't have the `mavlink_log_pub` you will not publish repeating messages as an in-between solution, right?

Comment (bkueng) [reply src/modules/commander/Commander.cpp]: Yes

**Comment by MaEtUgR** [comment]
btw same holds for other checks:
image

**Comment by dagar** [comment]
I'll do some testing and see if we can get this in a state to bring in for v1.11.

**Comment by LorenzMeier** [comment]
@MaEtUgR Do you have bandwidth to look into this today and potentially improve it? I'd like to get this in and not make it a too big change overall.

**Comment by LorenzMeier** [comment]
FYI: I'm closing in on this.

**Comment by LorenzMeier** [comment]
@MaEtUgR I've fixed the recurring reporting you've seen. It is all silent now and just reported via the GCS (as intended). From my perspective this is ready to go.

**Comment by LorenzMeier** [comment]
This is the output of `commander check` now. It is clear that we need wildly different flags to represent the systems a drone really cares about, but it is a good first step to consistently implement what MAVLink already has and to get the data paths structured and architected correctly:

```
pxh> commander check
INFO  [commander] Preflight check: OK
INFO  [commander] Prearm check: OK
INFO  [HealthFlags] DEVICE              STATUS
INFO  [HealthFlags] ----------------------------------
INFO  [HealthFlags] GYRO:               EN      OK
INFO  [HealthFlags] ACC:                EN      OK
INFO  [HealthFlags] MAG:                EN      OK
INFO  [HealthFlags] PRESS:              EN      OK
INFO  [HealthFlags] AIRSP:               
INFO  [HealthFlags] GPS:                EN      OK
INFO  [HealthFlags] OPT:                 
INFO  [HealthFlags] VIO:                 
INFO  [HealthFlags] LASER:               
INFO  [HealthFlags] GTRUTH:              
INFO  [HealthFlags] RATES:               
INFO  [HealthFlags] ATT:                 
INFO  [HealthFlags] YAW:                 
INFO  [HealthFlags] ALTCTL:              
INFO  [HealthFlags] POS:                 
INFO  [HealthFlags] MOT:                 
INFO  [HealthFlags] RC  :               EN      OFF
INFO  [HealthFlags] GYRO2:               
INFO  [HealthFlags] ACC2:                
INFO  [HealthFlags] MAG2:                
INFO  [HealthFlags] GEOFENCE:            
INFO  [HealthFlags] AHRS:               EN      OK
INFO  [HealthFlags] TERRAIN:             
INFO  [HealthFlags] REVMOT:              
INFO  [HealthFlags] LOGGIN:              
INFO  [HealthFlags] BATT:               EN      OK
INFO  [HealthFlags] PROX:                
INFO  [HealthFlags] SATCOM:              
INFO  [HealthFlags] PREARM:             EN      OK
INFO  [HealthFlags] OBSAVD:              
```

**Comment by hamishwillee** [comment]
1. Could we get module docs for `commander check` (i.e. use of PRINT_MODULE_DESCRIPTION) so this self-docs in http://dev.px4.io/master/en/middleware/modules_main.html ?

2. does `EN` mean device enabled?

3. What else does this do? It looks like maybe outputs the warnings on failures and hints as a stream of MAVLink STATUSTEXT messages?

**Pull request merged**